### PR TITLE
Enable user-as-a-case for a domain when it is used in an advanced form

### PR DESCRIPTION
**Pull request by kaapstorm**
Previously it was only enabled when it was used in a normal form.

@snopoke, buddy @dannyroberts

### CODE REVIEW

**Comment by kaapstorm** [comment]
[FB 172105](http://manage.dimagi.com/default.asp?172105)

File: corehq/apps/app_manager/views.py [hunk 1]
```diff
@@ -2049,6 +2050,8 @@ def edit_advanced_form_actions(request, domain, app_id, module_id, form_id):
     json_loads = json.loads(request.POST.get('actions'))
     actions = AdvancedFormActions.wrap(json_loads)
     form.actions = actions
+    if advanced_actions_use_usercase(form.actions) and not is_usercase_in_use(domain):
```
Comment (snopoke) [inline corehq/apps/app_manager/views.py]: the `is_usercase_in_user` check is redundant since that's already checked in `enable_usercase`

Comment (kaapstorm) [inline corehq/apps/app_manager/views.py]: You're right @snopoke. I'm happy to take it out of this function and `edit_form_actions()`. I'm wondering though, in `enable_usercase()` the `is_usercase_in_use` check is done inside a `CriticalSection`. Is that much of a performance hit? Should we be checking whether we need to call `enable_usercase()` first, or is it fine just to call it, and return if usercase is already enabled.

Comment (snopoke) [inline corehq/apps/app_manager/views.py]: Fine to leave as is

**Comment by dannyroberts** [comment]
@kaapstorm good to merge? Looks fine by me, feel free to self-merge when ready

**Pull request merged**